code/modules/hydroponics/plant_genes.dm

```diff
@@ -253,16 +253,31 @@
 	rate = 0.2
 
 /datum/plant_gene/trait/cell_charge/on_slip(obj/item/reagent_containers/food/snacks/grown/G, mob/living/carbon/C)
-	var/power = G.seed.potency*rate
+	var/power = round(G.seed.potency*rate)
 	if(prob(power))
-		C.electrocute_act(round(power), G, 1, 1)
+		var/offender = G.fingerprintslast
+		C.electrocute_act(power, G, 1, 1)
+		var/turf/T = get_turf(C)
+		if(C.ckey != offender)
+			C.investigate_log("[C] has slipped on an electric plant at [AREACOORD(T)] 	 last touched by [offender].", INVESTIGATE_BOTANY)
+			log_combat(C, G, "slipped on", null, "with the power of 10 last touched by [offender]")
 
 /datum/plant_gene/trait/cell_charge/on_squash(obj/item/reagent_containers/food/snacks/grown/G, atom/target)
 	if(iscarbon(target))
 		var/mob/living/carbon/C = target
 		var/power = G.seed.potency*rate
 		if(prob(power))
 			C.electrocute_act(round(power), G, 1, 1)
+			var/offender = G.fingerprintslast
+			if(C.ckey != offender)
+				var/mob/living/carbon/badperson
+				for(var/mob/M in GLOB.mob_living_list)
```

Comment:
Get the client by ckey from the associative list and then get their mob from that. (I dont know the lost name but I know it exists)